clients/da-vinci-client/src/main/java/com/linkedin/davinci/helix/AbstractPartitionStateModel.java

```diff
@@ -443,7 +421,88 @@ protected int getPartition() {
     return partition;
   }
 
+  protected String getStoreName() {
+    return storeName;
+  }
+
+  protected int getVersionNumber() {
+    return versionNumber;
+  }
+
   public String getStorePartitionDescription() {
     return storePartitionDescription;
   }
+
+  VeniceUserStoreType getStoreType() {
+    if (storeType == null) {
+      storeType = determineStoreType();
+    }
+    return storeType;
+  }
+
+  /**
+   * Returns a human-readable description of the replica type based on store type and version role.
+   * Examples: "System store future version", "Batch store current version", "Hybrid store backup version"
+   */
+  String getReplicaTypeDescription() {
+    VeniceUserStoreType type = getStoreType();
+    String role = getStoreVersionRole();
+
+    String storeTypeDesc;
+    if (type == VeniceUserStoreType.SYSTEM) {
+      storeTypeDesc = "System store";
```

Comment:
Do you think we could make 'description' string be part of the `VeniceUserStoreType` definition, so we don't have to have many `if-else` branches? Feel free to drop it, if it doesn't fit.